### PR TITLE
acid-state 0.16.1.1: Support mtl-2.3

### PR DESCRIPTION
**Pull request by andreasabel**
- add imports of `Control.Monad` to support `mtl-2.3`
- fix some warnings
- restore GHC 7.8 compatibility (one-liner in testsuite)
- bump base to >= 4.7 (was wrongly 4.5)
- CI for mtl-2.3 
- CHANGELOG

I lack the rights to upload a RC to hackage.
@adamgundry: Please add me to: https://hackage.haskell.org/package/acid-state/maintainers/

UPDATE (2022-06-01): candidate at https://hackage.haskell.org/package/acid-state-0.16.1.1/candidate

### CODE REVIEW

File: src/Data/Acid/Log.hs [hunk 1]
```diff
@@ -268,8 +272,8 @@ archiveFileLog :: FileLog object -> EntryId -> IO ()
 archiveFileLog fLog entryId = do
   logFiles <- findLogFiles (logIdentifier fLog)
   let sorted = sort logFiles
-      relevant = filterLogFiles Nothing (Just entryId) sorted \\
-                 filterLogFiles (Just entryId) (Just (entryId+1))  sorted
+      relevant = filterLogFiles Nothing (Just entryId) sorted
```
Comment (stepcut) [inline src/Data/Acid/Log.hs:275]: Is this just a stylistic choice or does it actually fix a compilation error? Not a complaint, just curious.

**Review by stepcut — APPROVED** [review]
Thanks so much!

File: src/Data/Acid/Log.hs (hunk 1 above)
Comment (andreasabel) [reply src/Data/Acid/Log.hs:275]: Yes, I was surprised myself that CPP could not handle a line ending in `\\`, neither on GHC 7.8 nor the latest GHC 9.2.
Moving it away from the end of the line fixed the problem.
(Note that I added LANGUAGE CPP to this file to get a conditional import.)

Comment (andreasabel) [reply src/Data/Acid/Log.hs:275]: Turns out that I reported this GHC bug 5 year ago already:
- https://gitlab.haskell.org/ghc/ghc/-/issues/12391

**Comment by andreasabel** [comment]
@stepcut @ddssff Re hackage upload privs:
David promised hackage upload privileges in https://github.com/acid-state/acid-state/issues/87#issuecomment-1045007693 . I argued there that this would be part of the privileges needed to be an effective stackage maintainer.  I think David forgot about it.  Please follow up on this an add me: https://hackage.haskell.org/package/acid-state/maintainers/

**Comment by adamgundry** [comment]
@andreasabel sorry for the delay on this, I've added you to the `acid-state` maintainer group on Hackage.

**Comment by andreasabel** [comment]
If you give me green light, I will publish the [candidate](https://hackage.haskell.org/package/acid-state-0.16.1.1/candidate).
The changes I did in this PR only require a patch-level version bump from 0.16.1 to 0.16.1.1.
@adamgundry : Does this also apply to the changes you did for `hedgehog`?

**Comment by andreasabel** [comment]
Published! https://hackage.haskell.org/package/acid-state-0.16.1.1

**Pull request merged**